Sparse matrix

DavZim opened the pull request:
Fixes #12. I.e., adds Sparse Matrix Support to solitude.

Notes:
- for sparse matrices, checks for duplicates are dropped (its more complicated to check duplicates in sparse matrices than it seems, apparently).
- as ranger accesses the data differently, we get slightly different results than when using a data.frame representation.

Test with this code:

```r
library(solitude)
library(Matrix)

data("humus", package = "mvoutlier")
columns_required = setdiff(colnames(humus)
                           , c("Cond", "ID", "XCOO", "YCOO", "LOI")
)
humus2 = humus[ , columns_required]

# sparse matrices wouldnt be too interesting in this case, but showcases the issue
humus_sp <- Matrix(as.matrix(humus2), sparse = TRUE)

set.seed(1)
index = sample(ceiling(nrow(humus_sp) * 0.5))
# initiate an isolation forest
iso = isolationForest$new(sample_size = length(index))
iso$fit(dataset = humus_sp[index, ])
iso$predict(humus_sp[index, ]) # scores for train data
iso$predict(humus_sp[-index, ]) # scores for new data (50% sample)
```

DavZim committed:
bump version

DavZim committed:
add sparse Matrix compatibility

talegari commented:
@DavZim Thanks for the PR. It looks good.

I have two change sugestions:

1.  `is_sparse = grepl("dg.Matrix", class(dataset)[[1]])` to `is_sparse = inherits(dataset, "dgCMatrix")`. Agree?
2. Add your name with email as contributor in DESCRIPTION file.

DavZim committed:
Add David to Description

DavZim commented:
Ideally I would like to use `inherits()`, the only issue is that there are multiple sparse matrix classes, e.g., `dgCMatrix`, `dgTMatrix`, `dgRMatrix`, ... (see also page 36ff https://cran.r-project.org/web/packages/Matrix/Matrix.pdf). Therefore, I introduced the `grepl()` call. Alternatively, we could do `mat_classes <- c("dgCMatrix", "dgTMatrix", ...); is_sparse <- any(sapply(mat_classes, inherits, x = dataset))`.
What do you think?

I've added myself as a contributor.

talegari commented:
From what I see in the documentation, ranger only supports `dgCMatrix` and none of the other types. We should test only for that class right?

DavZim commented:
Not totally sure on this. ranger will fail with its own message if the wrong type is provided, at the same time, we would support the other sparse matrix types if ranger decides to support them.

DavZim commented:
What are your thoughts on this? Should we merge this like it is, or do you want to provide error messages from solitude's side instead of letting ranger handle the data type compatibility?
